translator/trace/jaeger/protospan_to_jaegerproto.go

```diff
@@ -343,7 +343,13 @@ func ocMessageEventToJaegerTagsProto(msgEvent *tracepb.Span_TimeEvent_MessageEve
 
 // Replica of protospan_to_jaegerthrift appendJaegerTagFromOCSpanKind
 func appendJaegerTagFromOCSpanKindProto(jTags []jaeger.KeyValue, ocSpanKind tracepb.Span_SpanKind) []jaeger.KeyValue {
-	// We could check if the key is already present but it doesn't seem worth at this point.
+	// TODO: (@pjanotti): Replace any OpenTracing literals by importing github.com/opentracing/opentracing-go/ext?
```

Comment:
is not as simple as this but checking the attributes of the OC span avoids a possible (minor) perf pitfall: looping for all spans with a potentially large number of entries. It should be a relatively small hit but playing it defensively seems reasonable - in fact the map lookup can be slower than the loop for a small enough number of tags. That said the upper bound for the map lookup will be better. 

(Open to be convinced otherwise with some benchmarks to show that even for, let's say 100s, of tags this is not an issue.)

Reply:
I'm open to that - we would just have to update a few other places that check the list explicitly like: https://github.com/open-telemetry/opentelemetry-service/blob/master/translator/trace/jaeger/protospan_to_jaegerproto.go#L390-L394

We could change the call pattern to be more like how the jaeger thrift implementation was before my refactor and check the attribute map before invoking adding the a tag.

Reply:
> We could change the call pattern to be more like how the jaeger thrift implementation was before my refactor and check the attribute map before invoking adding the a tag.

It seems the less risky option.